src/index.js

```diff
@@ -50,7 +51,7 @@ function syncReduxAndRouter(history, store, selectRouterState = SELECT_STATE) {
     // Avoid dispatching an action if the store is already up-to-date,
     // even if `history` wouldn't do anything if the location is the same
     if(getRouterState().path !== locationToString(location)) {
-      store.dispatch(updatePath(locationToString(location)));
+      store.dispatch(updatePath(locationToString(location), true));
```

Comment:
Right, but that sounds like exactly how it has always worked. What is different now that `noRouterUpdate` is needed?

When the store listener is called, the paths will be the same already and nothing will happen. (I am traveling today so I won't be too responsive.)

Can you possibly remove this `true`, run it locally and tell me exactly what breaks?